docs/en/03_How_tos/external_http_requests_with_proxy.md

```diff
@@ -3,26 +3,35 @@ summary: How to access external resources via the CWP proxy.
 
 # External HTTP requests with proxy
 
-All CWP environments are positioned behind a proxy. All requests out to external services must go through this proxy.
+All requests to resources outside of your stack must use a proxy. CWP environments have no direct HTTP connectivity out to the internet nor to other CWP stacks.
 
-By default, we configure the PHP environment with `http_proxy` and `https_proxy` environment variables, as well as the `SS_OUTBOUND_PROXY` and `SS_OUTBOUND_PROXY_PORT` constants. These environment variables can be used within your code to send requests through the proxy.
+## Explicit configuration
 
-If you are seeing issues connecting to an external service, double check to make sure you are going through the proxy
-(compare against the constants `SS_OUTBOUND_PROXY` and `SS_OUTBOUND_PROXY_PORT`).
+To explicitly configure the proxy when making external HTTP requests, use the `SS_OUTBOUND_PROXY` and `SS_OUTBOUND_PROXY_PORT` PHP constants provided. Configuration of the outgoing proxy is dependent on the library you are using, so please consult relevant documentation.
 
-Because of this proxy, outgoing requests have a different IP than configured for incoming requests to your CWP hostname.
+For example curl can be configured by setting the `CURLOPT_PROXY` option via `curl_setopt` before making the actual request.
+
+If your library does not expose proxy configuration, you might be able to force it by configuring `http_proxy` and `https_proxy` environment variables through PHP's `putenv`. See [`InitialisationMiddleware`](https://github.com/silverstripe/cwp-core/blob/2/src/Control/InitialisationMiddleware.php) for a more specific code example.
+
+## Special case: curl from request handlers
+
+Plain `curl` requests will work out of the box, as long as you are making them from within request handlers (this might work for other libraries too).
+
+This is because the [`InitialisationMiddleware`](https://github.com/silverstripe/cwp-core/blob/2/src/Control/InitialisationMiddleware.php) sets the `http_proxy` and `https_proxy` environment variables automatically during the request execution.
+
+For requests made outside of request handling pipeline, you have to fall back to explicit library configuration.
+
+## Originating IP
+
+Outgoing requests originate at the proxy. This means they have an origin IP different from your stack's IP.
+
+If third party providers require whitelisting of IPs (for example on API requests performed through PHP on a CWP server), use CWP global proxy origin IPs:
 
-If third party providers require whitelisting of IPs (for example on API requests performed through PHP on a CWP server),
-please add the following IP addresses: 
 * 202.55.102.136 
 * 202.55.96.178
```

Comment:
Do we want to add clarify the clusters and the other cluster IPs as well?
Namely:
220.247.132.110
103.253.48.27
220.247.133.24
103.253.48.56

Reply:
Oh yes!